arangod/IResearch/IResearchViewBlock.cpp

```diff
@@ -69,7 +69,7 @@ pks_t::iterator readPKs(
   auto end = keys.end();
 
   for (irs::bytes_ref key; begin != end && it.next(); ) {
-    if (values(it.value(), key) && begin->read(key)) {
+    if (values(it.value(), key) && arangodb::iresearch::DocumentPrimaryKey::read(*begin, key)) {
```

Comment:
In this loop, `it` will always be advanced for every iteration, but `begin` will only be advanced conditionally, meaning the two can get out of sync.
That may be exactly what was intended here, but I don't know the exact intention.

Reply:
yes, that's exactly what is intended. The idea is to have a range of valid PKs in preallocated vector without gaps.